framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -128,6 +158,29 @@ PatternedMeshGenerator::generate()
   if (_z_width == 0)
     _z_width = bbox.max()(2) - bbox.min()(2);
 
+  std::vector<boundary_id_type> stitch_bids;
+  if (have_common_ids)
+    stitch_bids = input_bids[0];
+  else
+  {
+    for (boundary_id_type id = 0; id != Moose::INVALID_BOUNDARY_ID; ++id)
+      if (!used_boundary_ids.count(id))
+      {
+        stitch_bids.push_back(id);
+        if (stitch_bids.size() == 4)
+          break;
+      }
```

Comment:
I think there's a util to get the next free boundary id. and if there is not we should add one

Reply:
We use it in a few places but it's used in different ways, for example in RenameBoundaryGenerator we have to also check again an extra set of bids